LiteCore/Query/SQLiteDataFile+Indexes.cc

```diff
@@ -42,11 +42,15 @@ namespace litecore {
         if (indexTableExists())
             return;
 
+        if (!options().upgradeable)
+            error::_throw(error::CantUpgradeDatabase,
+                          "Accessing indexes requires upgrading the database schema");
```

Comment:
This is not usually valid.  It's only valid to throw this error if we are opening an older DB (2xx level).  In the usual case, we've created a 301+ database but the table doesn't exist yet.